pyiceberg/table/update/__init__.py

```diff
@@ -466,6 +466,21 @@ def _(update: SetSnapshotRefUpdate, base_metadata: TableMetadata, context: _Tabl
     return base_metadata.model_copy(update=metadata_updates)
 
 
+@_apply_table_update.register(RemoveSnapshotRefUpdate)
+def _(update: RemoveSnapshotRefUpdate, base_metadata: TableMetadata, context: _TableMetadataUpdateContext) -> TableMetadata:
+    if (existing_ref := base_metadata.refs.get(update.ref_name, None)) is None:
```

Comment:
I think this is a more Pythonic approach:
```suggestion
    if update.ref_name not in base_metadata.refs:
```

Reply:
I don't mind, I believe it was copied from https://github.com/apache/iceberg-python/pull/822